var/spack/repos/builtin/packages/moab/package.py

```diff
@@ -58,6 +58,7 @@ class Moab(Package):
     depends_on('mpi')
     depends_on('hdf5+mpi')
     depends_on('netcdf+mpi', when='+netcdf')
```

Comment:
since `mpi` is unconditional, i think this is a perfectly fine line. At this point, IMO there is no need making `mpi` a variant and then you may or may not need `depends_on('netcdf')`.  I would leave it as-is.